docs/api/java/getting-started.md

```diff
@@ -107,173 +90,190 @@ public class Main {
-                nonce,
-                fee.getErgsLimitNumber(),
-                estimate.getTo(),
-                estimate.getValueNumber(),
-                estimate.getData(),
-                fee.getMaxPriorityFeePerErgNumber(),
-                fee.getErgsPriceLimitNumber(),
-                signer.getAddress(),
-                meta
-        );
-
-        String signature = signer.getDomain().thenCompose(domain -> signer.signTypedData(domain, transaction)).join();
-        byte[] message = TransactionEncoder.encode(transaction, TransactionEncoder.getSignatureData(signature));
-
-        String sentTransactionHash = zksync.ethSendRawTransaction(Numeric.toHexString(message)).send().getTransactionHash();
-
-        // You can check the transaction status
-        TransactionReceipt receipt = zksync.ethGetTransactionReceipt(sentTransactionHash).send().getTransactionReceipt();
-    }
+  public static void main(String... args) {
+    ZkSync zksync; // Initialize client
+    Credentials credentials = Credentials.create("PRIVATE_KEY"); // Initialize signer
+    BigInteger chainId = zksync.ethChainId().send().getChainId();
+    EthSigner signer = new PrivateKeyEthSigner(credentials, chainId);
+    ZkSyncTransactionReceiptProcessor processor = new ZkSyncTransactionReceiptProcessor(zksync, DEFAULT_POLLING_FREQUENCY, DEFAULT_POLLING_ATTEMPTS_PER_TX_HASH);
+
+    String binary = "0x<bytecode_of_the_contract>";
+
+    byte[] salt = SecureRandom.getSeed(32);
+
+    // Here we can precompute contract address before its deploying
+    String precomputedAddress = ContractDeployer.computeL2Create2Address(new Address(signer.getAddress()), Numeric.hexStringToByteArray(binary), new byte[]{}, salt).getValue();
+
+    BigInteger nonce = zksync
+      .ethGetTransactionCount(credentials.getAddress(), DefaultBlockParameterName.PENDING).send()
+      .getTransactionCount();
+
+    Transaction estimate = Transaction.create2ContractTransaction(
+      credentials.getAddress(),
+      BigInteger.ZERO,
+      BigInteger.ZERO,
+      binary,
+      "0x",
+      salt
+    );
+
+    ZkTransactionFeeProvider feeProvider = new DefaultTransactionFeeProvider(zksync, Token.ETH);
+
+    Transaction712 transaction = new Transaction712(
+      chainId.longValue(),
+      nonce,
+      feeProvider.getGasLimit(estimate),
+      estimate.getTo(),
+      estimate.getValueNumber(),
+      estimate.getData(),
+      BigInteger.valueOf(100000000L),
+      feeProvider.getGasPrice(),
+      credentials.getAddress(),
+      estimate.getEip712Meta()
+    );
+
+    String signature = signer.getDomain().thenCompose(domain -> signer.signTypedData(domain, transaction)).join();
+    byte[] message = TransactionEncoder.encode(transaction, TransactionEncoder.getSignatureData(signature));
+
+    EthSendTransaction sent = zksync.ethSendRawTransaction(Numeric.toHexString(message)).send();
+
+    TransactionReceipt receipt = processor.waitForTransactionReceipt(sent.getResult());
```

Comment:
no try catch  or throw o waitFor..

Reply:
The above code also have a few unused imports

import io.zksync.methods.request.Eip712Meta;
import io.zksync.protocol.core.ZkBlockParameterName;
import io.zksync.transaction.fee.Fee;